lang/c++/impl/avrogencpp.cc

```diff
@@ -875,49 +872,104 @@ static string readGuard(const string &filename) {
     return candidate;
 }
 
+struct ProgramOptions {
+    bool helpRequested = false;
+    bool versionRequested = false;
+    bool noUnionTypedef = false;
+    std::string includePrefix = "avro";
+    std::string nameSpace;
+    std::string inputFile;
+    std::string outputFile;
+};
+
+static void printUsage() {
+    std::cout << "Allowed options:\n"
+              << "  -h [ --help ]                       produce help message\n"
+              << "  -V [ --version ]                    produce version information\n"
+              << "  -p [ --include-prefix ] arg (=avro) prefix for include headers, - for none, default: avro\n"
+              << "  -U [ --no-union-typedef ]           do not generate typedefs for unions in records\n"
+              << "  -n [ --namespace ] arg              set namespace for generated code\n"
+              << "  -i [ --input ] arg                  input file\n"
+              << "  -o [ --output ] arg                 output file to generate\n";
+}
+
+static bool parseArgs(int argc, char **argv, ProgramOptions &opts) {
+    for (int i = 1; i < argc; ++i) {
+        std::string arg = argv[i];
+
+        if (arg == "-h" || arg == "--help") {
+            opts.helpRequested = true;
+            return true;
+        }
+
+        if (arg == "-V" || arg == "--version") {
```

Comment:
Should we also check if there are any more arguments? For example, if someone type `avrogencpp -V -U`, should we just show the version or error out?

Reply:
`avrogencpp -V -U` prints the version without my change. So I think it is consistent with the original behavior.